### PR TITLE
feat(tls): enable specification of tls parameters

### PR DESCRIPTION
**Pull request by pree**
Enable the following parameters for the vault agent `vault` stanza:
* `ca_cert`
* `ca_path`
* `tls_server_name`
* `tls_skip_verify`

### CODE REVIEW

File: ansible/roles/vault-raft-backup-agent/templates/etc/vault.d/vault_snapshot_agent.hcl.j2 [hunk 1]
```diff
@@ -4,6 +4,10 @@ pid_file = "{{ vault_snapshot_pid_dir }}/{{ vault_snapshot_pid_file_name }}"
 
 vault {
   address = "{{ vault_tls_disable | ternary('http', 'https') }}://{{ vault_address }}:8200"
+  {%- if vault_ca_cert -%}ca_cert = "{{ vault_ca_cert }}"{% endif %}
+  {%- if vault_ca_path -%}ca_path = "{{ vault_ca_path }}"{% endif %}
+  {%- if vault_tls_server_name -%}tls_server_name = "{{ vault_tls_server_name }}"{% endif %}
+  tls_skip_verify = "{{ vault_tls_skip_verify | ternary('true', 'false') }}"
```
Comment (eyenx) [inline ansible/roles/vault-raft-backup-agent/templates/etc/vault.d/vault_snapshot_agent.hcl.j2:10]: this works with a value like "no" ?

Comment (tongpu) [reply ansible/roles/vault-raft-backup-agent/templates/etc/vault.d/vault_snapshot_agent.hcl.j2:10]: Yes, because Ansible. ;)

**Review by eyenx — APPROVED** [review]

**Comment by tongpu** [comment]
Did you test that this actually works as expected? I had some crazy munged lines in the output when testing it, which is the reason that I didn't approve it yet.

**Comment by pree** [comment]
yes, @in0rdr also tested it successfully. What errors/output did you get?

**Comment by tongpu** [comment]
I had `address`, `tls_skip_verify` and the new `tls_*` parameters all on the same line on my local machine. But I already suspected that my Ansible was somehow acting strange.

**Comment by in0rdr** [comment]
@tongpu I had the same experience when using @pree original code. But my suggestion in the code above fixed this behavior and I tested it thoroughly with all combinations of the different parameters. So it's not just your Ansible, it was the same behavior on my Ansible ;-) But if we change the whitespace control characters "-" as per my suggestion above it should render out just fine. Let me create a follow-up PR with the details.

**Comment by in0rdr** [comment]
https://github.com/adfinis-sygroup/vault-raft-backup-agent/pull/11

**Pull request merged**